src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -281,6 +278,8 @@ private bool HasNameInMetadata
             out bool isBad)
         {
             var typeWithModifiers = TypeSymbolWithAnnotations.Create(type, CSharpCustomModifier.Convert(customModifiers));
+            typeWithModifiers = typeWithModifiers.SetUnknownNullabilityForReferenceTypesIfNecessary(moduleSymbol);
```

Comment:
>SetUnknownNullabilityForReferenceTypesIfNecessary [](start = 50, length = 49)

It seems strange to create a `TypeSymbolWithAnnotations` then immediately potentially redo the work to set unknown nullability. Could it be done in a single go?

Reply:
Agreed, this should be optimized. Added `// PROTOTYPE` comment.

---
In reply to: [171076842](https://github.com/dotnet/roslyn/pull/25087#discussion_r171076842) [](ancestors = 171076842)